### PR TITLE
Finds a User's token that has the `admin:org_hook` scope

### PR DESCRIPTION
**Pull request by BenEmdon**
This PR proposes methods for `OrganizationWebhook` to find User tokens with the `admin:org_hook` scope. These tokens are required to create webhooks for organizations.

In a follow up PR I will make the methods for creating the webhooks using these tokens.

Part of #1647 
> - [ ] move the logic for creating a webhook from the `Organization::Creator` to `GitHubOrganizationWebhook`

### CODE REVIEW

File: app/models/organization_webhook.rb [hunk 1]
```diff
@@ -9,7 +11,62 @@ class OrganizationWebhook < ApplicationRecord
...
+  #
+  # Returns a User's token or raises a NoValidTokenError if one could not be found.
+  #
+  # Warning: This could potentially take very long for organizations
+  # of a large size, so invoke cautiously.
+  def admin_org_hook_scoped_github_client
+    token = if Rails.env.test?
+              users.first&.token
+            else
+              user_with_admin_org_hook_scope.sample&.token
```
Comment (srinjoym) [inline app/models/organization_webhook.rb]: Should this be calling the method underneath? There's a small typo
```suggestion
              users_with_admin_org_hook_scope.sample&.token
```

File: spec/models/organization_webhook_spec.rb [hunk 2]
```diff
@@ -18,18 +18,48 @@
...
+      end
     end
+  end
 
-    context "has no organizations" do
-      it "returns nil" do
-        expect(subject)
-          .to receive(:organizations)
+  describe "#users_with_admin_org_hook_scope" do
+    context "user with admin_org hook scope exists doesn't exist" do
```
Comment (srinjoym) [inline spec/models/organization_webhook_spec.rb]: Test name typo? :
```suggestion
    context "user with admin_org hook scope doesn't exist" do
```

File: spec/models/organization_webhook_spec.rb [hunk 3]
```diff
@@ -18,18 +18,48 @@
...
-    it "requests a GitHubOrganization" do
-      expect_any_instance_of(Organization).to receive(:github_organization)
-      subject.github_organization
+  describe "#admin_org_hook_scoped_github_client" do
+    context "token is present" do
+      before do
+        allow(subject).to receive_message_chain(:users, :first, :token) { "token" }
+      end
+
+      it "raises a NoValidTokenError" do
```
Comment (srinjoym) [inline spec/models/organization_webhook_spec.rb]: We don't expect to raise an error here, just need to update test name?

Comment (BenEmdon) [reply spec/models/organization_webhook_spec.rb]: Good catch 👍 a case of some copy pasta 🍝

Comment (BenEmdon) [inline spec/models/organization_webhook_spec.rb]:
```suggestion
      it "returns a Octokit::Client" do
```

**Pull request merged**